### PR TITLE
21174 Make the Particle Emitter Easier to Use

### PR DESCRIPTION
**Pull request by andrewd440**
For: https://worklist.net/21174

**Functionality/Code Changes:**
- Moves the qml Edit tabview to separate file to allow function usage of fromScript() to respond to script messages.
- Shows the particle explorer when selecting a particle entity.
- Replaces color picker used in particle explorer with the one used in entity properties.
- Always shows hand lasers while in edit mode.
- Allows selecting particle entities with hand lasers while in edit mode.

**Test Plan:**
- Enter VR mode with HMD and hand controllers active
- Creating a particle entity should show the particle explorer tab.
- Selecting a particle entity in the entity list should show the particle explorer tab.
- Selecting a particle entity in edit mode with the hand lasers should show the particle explorer tab.
- Particle entities should be selectable with hand lasers by selecting the particle overlay icon.
- The color picker used for the particle explorer should match the one used on the entity properties tab.
- When in edit mode and the stylus/finger is not active for the tablet, hand lasers should be shown.

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
An admin must verify this patch for the pull request build to fire. Type "ok to test" to begin testing this pull request or "add to whitelist" to test this and all future pull requests from this collaborator.

**Comment by cozza13** [comment]
ok to test

**Comment by cozza13** [comment]
add to whitelist

**Comment by hifi-gustavo** [comment]
Build console output for commit: 5ffb8eba868a23c943fb2961b07a12ac6da1f065 available at: https://gist.github.com/cc294eaef718895a4edbec7cacb3aa2f<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR10279-5ffb8eb.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR10279-5ffb8eb.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 5ffb8eba868a23c943fb2961b07a12ac6da1f065 available at: https://gist.github.com/cc294eaef718895a4edbec7cacb3aa2f<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR10279-5ffb8eb.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR10279-5ffb8eb.dmg

**Comment by Nex-Pro** [comment]
Testing this win10 and Vive

**Comment by Nex-Pro** [comment]
### PASS

Tested with win10 and Vive.

Compared against master where the color pickers are almost useless.

Test Plan:

Enter VR mode with HMD and hand controllers active

Creating a particle entity should show the particle explorer tab. **SUCCESS**

Selecting a particle entity in the entity list should show the particle explorer tab. **SUCCESS**

Selecting a particle entity in edit mode with the hand lasers should show the particle explorer tab. **SUCCESS**

Particle entities should be selectable with hand lasers by selecting the particle overlay icon. **NOT RELIABLE**  This is NOT due to this PR I see this issue in all builds. I may have to click many times to get a particle or light to select via the icon.

The color picker used for the particle explorer should match the one used on the entity properties tab. **SUCCESS**

When in edit mode and the stylus/finger is not active for the tablet, hand lasers should be  **SUCCESS**

File: interface/resources/qml/hifi/tablet/Edit.qml [hunk 1]
```diff
@@ -30,270 +22,10 @@ StackView {
...
-                    anchors.bottom: parent.bottom
-                    anchors.bottomMargin: 0
-                    anchors.top: parent.top
-                    anchors.topMargin: 0
-                }
-            }
-        }
-    }
+	// Passes script messages to the item on the top of the stack
+	function fromScript(message) {
```
Comment (zfox23) [inline interface/resources/qml/hifi/tablet/Edit.qml]: Four spaces instead of tabs throughout, please.

**Comment by zfox23** [comment]
Does this work on Desktop mode with toolbar?

If so, and the tabs -> spaces issue is fixed, I can dev-approve this.

**Comment by andrewd440** [comment]
Everything still works out on desktop mode from my testing, which the only that that changed is the color picker.

**Comment by zfox23** [comment]
Sweet, merged.

**Comment by hifi-gustavo** [comment]
Build console output for commit: d3a9114676da735d098957f4174a534829dce3fa available at: https://gist.github.com/cc294eaef718895a4edbec7cacb3aa2f<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR10279-d3a9114.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR10279-d3a9114.dmg

**Pull request merged**